src/Init/Data/Array/Subarray.lean

```diff
@@ -9,25 +9,34 @@ import Init.Data.Array.Basic
 universe u v w
 
 structure Subarray (α : Type u)  where
-  as : Array α
+  array : Array α
   start : Nat
   stop : Nat
-  h₁ : start ≤ stop
-  h₂ : stop ≤ as.size
+  start_le_stop : start ≤ stop
+  stop_le_array_size : stop ≤ array.size
+
+@[deprecated]
+def Subarray.as (s : Subarray α) : Array α := s.array
+
+@[deprecated]
+def Subarray.h₁ (s : Subarray α) : s.start ≤ s.stop := s.start_le_stop
+
+@[deprecated]
+def Subarray.h₂ (s : Subarray α) : s.stop ≤ s.array.size := s.stop_le_array_size
```

Comment:
```suggestion
@[deprecated Subarray.array]
abbrev Subarray.as (s : Subarray α) : Array α := s.array

@[deprecated Subarray.start_le_stop]
def Subarray.h₁ (s : Subarray α) : s.start ≤ s.stop := s.start_le_stop

@[deprecated Subarray.stop_le_array_size]
def Subarray.h₂ (s : Subarray α) : s.stop ≤ s.as.size := s.stop_le_array_size
```
so that the deprecation tells you what to use instead

Reply:
Thanks! I was trying to figure this one out.